### DISCUSSION TITLE
Attitude estimation when attitude is manually changed

### DISCUSSION THREAD

**Discussion by GNC-JY** [bitcraze/discussions#739] [Q&A] [answered]
![image](https://github.com/bitcraze/discussions/assets/110527007/cfaa2cd1-bca6-4809-86a7-d65fa774c654)
Hello.
I'm using the crazyflie 2.1 and Firmware ver. 2023.02 and I'm not detecting any decks.
Only Pitch rotation is possible by tying it to both holes of the Crazyflie with thread.
When the pitch of the started Crazyflie is arbitrarily moved, the Attitude Estimate values of Roll and Yaw tend to move at the same time as shown in the figure.
I am in the process of continuing to analyze Complementary Filter Code, and the process is difficult.
Can you tell me why this reaction is happening?
And when I let go of my hand after moving the Pitch, the attitude recovers, but there is a problem with Roll and Yaw, the m2 motor turns off and the combined value of m1 and m4 is equal to m3.
![image](https://github.com/bitcraze/discussions/assets/110527007/d655f5bd-40ce-4273-a4da-398f0a9aeaac)

Why is that? I wonder what you think.

**Comment by knmcguire** [comment, marked as answer]
Hmm interesting! Does this happen with the kalman filter as well?

And exactly which logs variables are we looking at? there are [multiple logging variables](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/) available for roll pitch and yaw that mean different things

**Reply by GNC-JY** [reply]
Kalman filter also same response.
That parameters came from stateEstimate roll, pitch and yaw value.

**Reply by GNC-JY** [reply]
The part where the roll moved when the pitch was moved was solved through angle correction. However, I still don't know about the situation where the motor is turned off.
![image](https://github.com/bitcraze/discussions/assets/110527007/aa418f6e-d9ea-4fb9-8d28-290323b4a36c)

This is a motor out graph when Crazyflie is rotating when the motor is turned off and then attempts to recover using assist mode.

**Reply by knmcguire** [reply]
Alright, so you figured out your first issue so that is good. 

But your second issue is about the motors? What is the issue here exactly as I don't seem to understand what the problem is. The crazyflie is able to correct itself right?

**Reply by GNC-JY** [reply]
In detail, for example, when the pitch rate is rotating at 720deg/s, when starting using assist mode (ex. Alt Hold), the motor may be out as shown in the picture above. Analysis shows that both attitude desired and actual are close to zero, but when looking at four motor out, the sum of the two front motor out is the same as the sum of the one rear motor out. And two motors in the clockwise direction are more thrust than two motors in the counterclockwise direction.

**Reply by knmcguire** [reply]
I'm not exactly sure what the problem is. The motors are reacting to the controller output and if indeed the pitch is rotated but the angle correction is giving the yaw and roll also a value, then I would expect that to be honest. Also, for a yaw motion, the higher spin of one direction compared to another is also to be as expected. 

So... You could look into the firmware with motor mixing a bit, but perhaps for the continuation of support for this, it would be best to explain what the goal exactly is.

**Reply by GNC-JY** [reply]
I did the analysis, and I think it is caused by the control structure problem when the attitude is recovered due to accumulation due to ki gain in Complementary filter. I should study the estimator in more depth.
